hack/app-sre/build-push.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Build and push the operator image for direct deploy (no OLM).
+
+set -exv
+
+CURRENT_DIR=$(dirname $0)
+
+BASE_IMG="hive"
+# This must correspond to the REGISTRY_IMG variable defined in the
+# app-interface saas file: data/services/hive/cicd/ci-int/saas-hive.yaml
+QUAY_IMAGE="quay.io/app-sre/${BASE_IMG}"
+IMG="${BASE_IMG}:latest"
+
+# This must correspond to the IMAGE_TAG generated by app-sre: As long as we
+# *don't* set `use_channel_in_image_tag: true` in the saas file, this is the
+# first $NUM digits of the current commit hash. $NUM is 7 by default (if we
+# don't override it by setting `hash_length` in the saas file).
+GIT_HASH=`git rev-parse --short=7 HEAD`
+
+# build the image
+BUILD_CMD="docker build" IMG="$IMG" make GO_REQUIRED_MIN_VERSION:= docker-build
```

Comment:
what was previous method of building hive and pushing the container image? just to compare that with this new flow??

Reply:
See [hack/app_sre_build_deploy.sh](https://github.com/openshift/hive/blob/0544ad7891b18e6ed34cd5ce39b98fb604e7ce8a/hack/app_sre_build_deploy.sh). This is a subset: the first part (build/push the operator image) but not the second part (build/push the OLM bundles).

Reply:
i propose to keep using the original script. when the time comes and we are off OLM - you can just remove the redundant section from the script.

Reply:
Right, we'll keep using the original for now (as updated in [!22093](https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/22093)). Eventually I'd like to move (so everything app-sre is under hack/app-sre) and rename (to reflect what it's actually doing). If someone feels strongly about it, I can get rid of this script from this PR.

Reply:
I removed it.